### PR TITLE
Improve code size limit tests

### PR DESCRIPTION
**Pull request by gumb0**
1. Add two new cases:
- Valid: creation transaction returning code with max allowed size (0x6000)
- Invalid: creation transaction returning code with size 0x6001

2. Huge initcode is not needed, because CODECOPY implicitly pads code with 0s, so I suggest to remove the code after RETURN for simplification.

3. (optional) I replaced the bytecode with yul. 

4. New tests similar to 1 for CREATE/CREATE2 opcodes

### CODE REVIEW

File: src/GeneralStateTestsFiller/stCodeSizeLimit/codesizeOOGInvalidSizeFiller.json [hunk 1]
```diff
@@ -33,8 +33,11 @@
             }
         },
         "transaction" : {
+            "// code size is 0x600d" :" mstore(0x40, 0x60) codecopy(0x00, 0x0012, 0x600d) return(0x00, 0x600d)",
```
Comment (gumb0) [inline src/GeneralStateTestsFiller/stCodeSizeLimit/codesizeOOGInvalidSizeFiller.json]: Note: it's rather pseudocode than exact disassembly, it actually uses DUP1 for code size constant.
(maybe would make sense to just replace raw bytecode with yul here?)

Comment (gumb0) [reply src/GeneralStateTestsFiller/stCodeSizeLimit/codesizeOOGInvalidSizeFiller.json]: Also I didn't get why MSTORE is needed, probably it's not.

Comment (marioevz) [reply src/GeneralStateTestsFiller/stCodeSizeLimit/codesizeOOGInvalidSizeFiller.json]: Just checked and indeed the MSTORE is not required.

File: src/GeneralStateTestsFiller/stCodeSizeLimit/createCodeSizeLimitFiller.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# CREATE with max allowed deployed code size
+createCodeSizeLimit:
```
Comment (winsvega) [inline src/GeneralStateTestsFiller/stCodeSizeLimit/createCodeSizeLimitFiller.yml:2]: very nice!

**Comment by winsvega** [comment]
This branch is out-of-date with the base branch
rebase?

**Review by winsvega — APPROVED** [review]

**Comment by gumb0** [comment]
Rebased.

**Pull request merged**